### PR TITLE
[Admin] Enable and Implement Remaining Scenarios

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A
|-----------------|-----
| Branch?         | 2.0 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no 
| License         | MIT

### CODE REVIEW

File: features/hybrid/promotion/applying_catalog_promotions/not_reapplying_catalog_promotions_on_variants_once_products_taxon_changes.feature [hunk 1]
```diff
@@ -16,14 +16,14 @@ Feature: Not reapplying catalog promotions on variants once the product’s taxo
         And there is disabled catalog promotion "Surprise sale" between "2021-07-01" and "2022-05-04" available in "Web-US" channel that reduces price by "90%" and applies on "Dishes" taxon
         And I am logged in as an administrator
 
-    @api @todo @ui
+    @api @ui @mink:chromedriver
```
Comment (Rafikooo) [inline features/hybrid/promotion/applying_catalog_promotions/not_reapplying_catalog_promotions_on_variants_once_products_taxon_changes.feature:19]: I encountered issues covering these scenarios without the JS driver, likely due to our current method of accessing Taxons with bootstrap inputs.

File: src/Sylius/Bundle/AdminBundle/templates/product/common/header/title_block/actions/view_in_store.html.twig [hunk 2]
```diff
@@ -1,6 +1,11 @@
 {% from '@SyliusAdmin/shared/helper/icon.html.twig' import icon %}
 
-{% set product = (hookable_metadata.context.resource) %}
+{% if constant('class', hookable_metadata.context.resource) == 'Sylius\\Component\\Core\\Model\\Product' %}
+    {% set product = hookable_metadata.context.resource %}
+{% else %}
+    {% set product = hookable_metadata.context.resource.product %}
+{% endif %}
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/templates/product/common/header/title_block/actions/view_in_store.html.twig]: What other resource can we have here?

File: src/Sylius/Behat/Page/Admin/Product/IndexPage.php [hunk 3]
```diff
@@ -113,9 +113,8 @@ protected function getDefinedElements(): array
             'first_product' => '.table > tbody > tr:first-child',
             'last_product' => '.table > tbody > tr:last-child',
             'main_taxon_filter' => '#criteria_main_taxon',
-            'page_number' => '.sylius-grid-nav__pagination .active',
-            'pagination_button' => '.sylius-grid-nav__pagination a.item:contains("%page%")',
-            'pagination_buttons' => '.sylius-grid-nav__pagination',
+            'current_page' => '[data-test-current-page]',
+            'page_button' => '[data-test-page="%page%"]',
```
Comment (NoResponseMate) [inline src/Sylius/Behat/Page/Admin/Product/IndexPage.php]: Shouldn't these be moved to the crud index page instead?

File: src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/product_variant/update.yaml [hunk 4]
```diff
@@ -13,6 +13,9 @@ twig_hooks:
         'sylius_admin.product_variant.update.content.header.title_block.actions':
             price_history:
                 template: '@SyliusAdmin/product_variant/form/content/update/header/actions/price_history.html.twig'
+            view_in_store:
+                template: '@SyliusAdmin/product/common/header/title_block/actions/view_in_store.html.twig'
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/product_variant/update.yaml]: No. We're outside of the `product` scope here.

File: src/Sylius/Bundle/AdminBundle/templates/product/common/header/title_block/actions/view_in_store.html.twig (hunk 2 above)
Comment (Wojdylak) [inline src/Sylius/Bundle/AdminBundle/templates/product/common/header/title_block/actions/view_in_store.html.twig]:
```suggestion
{% set product = hookable_metadata.context.resource.product|default(hookable_metadata.context.resource) %}
```

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @Rafikooo!

**Pull request merged**